src/discounts/views/DiscountDetails/DiscountDetails.tsx

```diff
@@ -1,35 +1,157 @@
+
+  const [promotionUpdate, promotionUpdateOpts] = usePromotionUpdateMutation({
+    onCompleted(data) {
+      if (data?.promotionUpdate?.errors?.length === 0) {
+        notify({
+          status: "success",
+          text: intl.formatMessage(commonMessages.savedChanges),
+        });
+        refetch();
+      }
+    },
+  });
+
+  const [promotionRuleUpdate, promotionRuleUpdateOpts] =
+    usePromotionRuleUpdateMutation({
+      onCompleted(data) {
+        if (data?.promotionRuleUpdate?.errors?.length === 0) {
+          notify({
+            status: "success",
+            text: intl.formatMessage(commonMessages.savedChanges),
+          });
+          refetch();
+        }
+      },
+    });
+
+  const [promotionRuleCreate, promotionRuleCreateOpts] =
+    usePromotionRuleCreateMutation({
+      onCompleted(data) {
+        if (data?.promotionRuleCreate?.errors?.length === 0) {
+          notify({
+            status: "success",
+            text: intl.formatMessage(commonMessages.savedChanges),
+          });
+          refetch();
+        }
+      },
+    });
+
+  const [promotionDelete, promotionDeleteOpts] = usePromotionRuleDeleteMutation(
+    {
+      onCompleted(data) {
+        if (data?.promotionRuleDelete?.errors?.length === 0) {
+          notify({
+            status: "success",
+            text: intl.formatMessage(commonMessages.savedChanges),
+          });
+          refetch();
+        }
+      },
+    },
+  );
+
+  const onSubmit = createUpdateHandler(promotionData?.promotion, variables =>
+    promotionUpdate({ variables }),
+  );
+
+  const onRuleUpdateSubmit = createRuleUpdateHandler(
+    promotionData?.promotion,
+    variables => promotionRuleUpdate({ variables }),
+  );
+
+  const onRuleCreateSubmit = createRuleCreateHandler(
+    promotionData?.promotion,
+    variables => promotionRuleCreate({ variables }),
+  );
+
+  const onRuleDeleteSubmit = (id: string) => {
+    return promotionDelete({
+      variables: {
+        id,
+      },
+    });
+  };
+
   return (
     <>
       <WindowTitle title={intl.formatMessage(commonMessages.discounts)} />
       <DiscountDetailsPage
-        disabled={false}
+        data={promotionData?.promotion}
+        errors={getMutationErrors(promotionUpdateOpts)}
+        disabled={loading || promotionUpdateOpts.loading}
+        ruleConditionsOptionsDetailsMap={getRuleConditionsOptionsDetailsMap(
+          ruleConditionsOptionsDetails,
+        )}
+        ruleConditionsOptionsDetailsLoading={
+          ruleConditionsOptionsDetailsLoading
+        }
         onBack={() => {
-          navigate(saleListUrl());
+          navigate(discountListUrl());
         }}
         channels={availableChannels}
-        discount={discount}
-        onSubmit={() => {}} // To be implemented
-        onRuleSubmit={() => {}} // To be implemented
+        onSubmit={onSubmit}
+        submitButtonState={promotionUpdateOpts.status}
+        onRuleUpdateSubmit={onRuleUpdateSubmit as any}
```

Comment:
**issue** casting as any should not be used, this can be typed using `ReturnType<typeof onRuleUpdateSubmit>` in the child component or something similar